docs/source/usingmame/defaultkeys.rst

```diff
@@ -1,106 +1,558 @@
+
+**5**
+    Coin slot 1
+**6**
+    Coin slot 2
+**7**
+    Coin slot 3
+**8**
+    Coin slot 4
+**1**
+    Player 1 start or 1 player mode
+**2**
+    Player 2 start or 2 players mode
+**3**
+    Player 3 start or 3 players mode
+**4**
+    Player 4 start or 4 players mode
+**Backspace**
+    Bill 1 (For machines that have a bill receptor/note reader)
+**9**
+    Service 1 (Service buttons are typically used to give free credits or to
+    navigate the internal operator service menus)
+**0**
+    Service 2
+**-** (not numeric keypad)
+    Service 3
+**=** (not numeric keypad)
+    Service 4
+**Up Arrow**
+    Player 1 Up
+**Down Arrow**
+    Player 1 Down
+**Left Arrow**
+    Player 1 Left
+**Right Arrow**
+    Player 1 Right
+**E**
+    Up on Left Stick for dual-stick machines (e.g. Robotron)
+**D**
+    Down on Left Stick for dual-stick machines (e.g. Robotron)
+**S**
+    Left on Left Stick for dual-stick machines (e.g. Robotron)
+**F**
+    Right on Left Stick for dual-stick machines (e.g. Robotron)
+**I**
+    Up on Right Stick for dual-stick machines (e.g. Robotron)
+**K**
+    Down on Right Stick for dual-stick machines (e.g. Robotron)
+**J**
+    Left on Right Stick for dual-stick machines (e.g. Robotron)
+**L**
+    Right on Right Stick for dual-stick machines (e.g. Robotron)
+**Ctrl**/**Mouse B0**/**Gun 1 Button 0**
+    Player 1 Button 1
+**Alt**/**Mouse B2**/**Gun 1 Button 1**
+    Player 1 Button 2
+**Spacebar**/**Mouse B1**/**Joystick 1 Button 1 or B**
+    Player 1 Button 3
+**Shift**
+    Player 1 Button 4
+**Z**
+    Player 1 Button 5
+**X**
+    Player 1 Button 6
+**C**
+    Player 1 Button 7
+**V**
+    Player 1 Button 8
+**B**
+    Player 1 Button 9
+**N**
+    Player 1 Button 10
+**M**
+    Player 1 Button 11
+**,**
+    Player 1 Button 12
+**.**
+    Player 1 Button 13
+**/**
+    Player 1 Button 14
+**Right Shift**
+    Player 1 Button 15
+**9**
+    Service
+**0**
+    Book-Keeping
+
+
+.. _default-mahjong-hanafuda-keys:
+
+Default Mahjong and Hanafuda Keys
+=================================
+
+All the keys below are fully configurable in the user interface. This list shows
+the standard keyboard configuration.
+
+
+**5**
+    Coin slot 1
+**6**
+    Coin slot 2
+**7**
+    Coin slot 3
+**8**
+    Coin slot 4
+**1**
+    Player 1 start or 1 player mode
+**2**
+    Player 2 start or 2 players mode
+**3**
+    Player 3 start or 3 players mode
+
+    Mahjong Bet
+**4**
+    Player 4 start or 4 players mode
+**Backspace**
+    Bill 1 (For machines with a bill receptor, typically gambling)
+**9**
+    Service 1 (Service buttons are typically used to give free credits or to
+    navigate the internal operator service menus)
+**0**
+    Service 2
+**-** (not numeric keypad)
+    Service 3
+**=** (not numeric keypad)
+    Service 4
+**Ctrl**/**Mouse B0**/**Gun 1 Button 0**
+    Mahjong Kan
+**Alt**/**Mouse B2**/**Gun 1 Button 1**
+    Mahjong Pon
+**Spacebar**/**Mouse B1**/**Joystick 1 Button 1**
```

Comment:
Are you actually getting joystick/mouse/gun defaults on these?  From a quick look at the source, it’s only got the keyboard inputs there.  A quick run of `hotgmck` seemed to confirm this.

Reply:
Oddly, it did once with a controller file that didn't map any Mahjong buttons when I first loaded one up to start pulling mappings, but I can't reproduce it with a completely clean set.

I'll remove those shortly.